dialectic/src/unary.rs

```diff
@@ -176,12 +168,40 @@ where
     type Result = S<<(N, M) as Add>::Result>;
 }
 
+/// A trait marking wrapped type-level constants.
+pub trait Constant: sealed::Constant {}
+
+/// A wrapper for type-level `usize` values to allow implementing traits on them.
+#[allow(missing_debug_implementations)]
```

Comment:
Why not add a `Debug` implementation? Indeed, this can trivially implement every derivable trait.

Reply:
`Number` shouldn't really show up anywhere since it's a marker/wrapper type essentially that isn't supposed to be used as anything but a vehicle for trait resolution. I figured that leaving it not implementing anything would better raise red flags if someone tries to use it as a type parameter. Thoughts?

Reply:
"missing `Debug` implementation for `Number<0>`" doesn't exactly say "this is an internal-only type" to me, and that's what the user would see. Why not instead name `Number` an empty enum, but implement all the traits vacuously? Does that work?

Reply:
Hmm, yeah, that should work. Alright, I'll take care of that.